cpp/include/cudf/detail/distinct_hash_join.cuh

```diff
@@ -42,6 +42,9 @@ template <typename Equal>
 struct comparator_adapter {
   comparator_adapter(Equal const& d_equal) : _d_equal{d_equal} {}
 
+  // suppress "function was declared but never referenced warning"
```

Comment:
`function was declared but never referenced` means we never use these functions. Is this the correct diagnosis? If so can we remove these functions?

Reply:
Yeah, the intention is a bit obscure here. 

Basically, all functions involved in this PR have two overloads: one is working with the current old cuco and the other will work with the upcoming new cuco. Only one of those two will be used at the same point thus the other is always redundant. I will remove redundant ones once we migrate to use the new cuco. The goal of this work is mainly to avoid breaks when `rapids-cmake` fetches the new cuco but libcudf can only work with the old cuco.

Reply:
That makes sense. Thank you.